src/app/modules/form/form.service.ts

```diff
@@ -117,3 +119,35 @@ export const isFormPublic = (
       return err(new PrivateFormError(form.inactiveMessage, form.title))
   }
 }
+
+/**
+ * Method to check whether a form has reached submission limits, and deactivate the form if necessary
+ * @param form the form to check
+ * @returns ok(true) if submission is allowed because the form has not reached limits
+ * @returns ok(false) if submission is not allowed because the form has reached limits
+ */
+export const checkFormSubmissionLimitAndDeactivateForm = async (
+  form: IPopulatedForm,
+): Promise<Result<true, PrivateFormError>> => {
+  if (form.submissionLimit !== null) {
+    const currentCount = await SubmissionModel.countDocuments({
+      form: form._id,
+    }).exec()
+
+    if (currentCount >= form.submissionLimit) {
+      logger.info({
+        message: 'Form reached maximum submission count, deactivating.',
+        meta: {
+          form: form._id,
+          action: 'checkFormSubmissionLimitAndDeactivate',
+        },
+      })
+      await deactivateForm(form._id)
+      return err(new PrivateFormError(form.inactiveMessage, form.title))
```

Comment:
create a new Error type?

Reply:
Again, I think the point is that this should be indistinguishable from a deactivated form?

Reply:
I see, yeah you're right, this works.